eng/pipelines/templates/stages/archetype-sdk-tool-dotnet.yml

```diff
@@ -102,11 +96,33 @@ stages:
             artifact: packages
             condition: succeededOrFailed()
 
-          - ${{ if gt(length(parameters.StandaloneExeMatrix), 0) }}:
-              - publish: $(Build.ArtifactStagingDirectory)/binaries
-                displayName: Publish executables to binaries artifact
-                artifact: binaries
-                condition: succeededOrFailed()
+      - job: Produce_Executables
```

Comment:
@scbedd did you intend this job to exist for all tools even if they don't have executables? I noticed this was running in another pipeline and doing no real work because StandaloneExeMatrix  was empty. Perhaps we should try to exclude these jobs if they aren't building executables.